### ISSUE TITLE
org.mybatis.generator.plugins.ToStringPlugin: missing key column in model's toString method

### ISSUE THREAD

**Issue by caladyon** [mybatis/generator#53]
Greetings,

when dealing with a table with non-primitive types (generating two classes: the first with primary-key columns only, and the second with the other columns), the toString() method generated for the second class misses the fields of the first class.

**Example**
As you can see, Tbit88Hp.toString() does not use the cSeg, cTypStr, nInt fields.

Setup: postgres 9.1, mybatis 3.3.0, MBG 1.3.3-SNAPSHOT

The Table:

``` sql
CREATE TABLE bita.tbit88_hp
(
  c_seg bigint NOT NULL,
  c_typ_str character varying(3) NOT NULL,
  n_int integer NOT NULL,
  q_spd real NOT NULL,
  q_est_var real NOT NULL,
  q_rlb real NOT NULL,
  d_lst_upd timestamp(0) with time zone,
  CONSTRAINT tbit88_hp_pkey PRIMARY KEY (c_seg, c_typ_str, n_int)
)
WITH (
  OIDS=FALSE
);
```

MBG configuration:

``` xml
<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE generatorConfiguration PUBLIC "-//mybatis.org//DTD MyBatis Generator Configuration 1.0//EN" "http://mybatis.org/dtd/mybatis-generator-config_1_0.dtd" >
<generatorConfiguration>

    <context id="development">
        <plugin type="org.mybatis.generator.plugins.SerializablePlugin"></plugin>
        <plugin type="org.mybatis.generator.plugins.RowBoundsPlugin"></plugin>
        <plugin type="org.mybatis.generator.plugins.ToStringPlugin"></plugin>
        <plugin type="org.mybatis.generator.plugins.VirtualPrimaryKeyPlugin"></plugin>

        <jdbcConnection driverClass="${dev.db.driver}"
            connectionURL="${dev.db.url}" userId="${dev.db.username}" password="${dev.db.password}" />

        <javaModelGenerator targetPackage="${generated.model.package}"
            targetProject="${project.source.dir}">
            <property name="enableSubPackages" value="true" />
            <property name="trimStrings" value="true" />
        </javaModelGenerator>

        <sqlMapGenerator targetPackage="${generated.mapper.package}"
            targetProject="${project.resources.dir}">
            <property name="enableSubPackages" value="true" />
        </sqlMapGenerator>

        <javaClientGenerator targetPackage="${generated.mapper.package}"
            targetProject="${project.source.dir}" type="XMLMAPPER">
            <property name="enableSubPackages" value="true" />
        </javaClientGenerator>

        <table schema="bita" tableName="tbit88_hp"
            enableSelectByPrimaryKey="true" 
            enableSelectByExample="true" 
            enableCountByExample="true" 
            enableInsert="true"
            enableUpdateByPrimaryKey="true" 
            enableUpdateByExample="true"
            enableDeleteByPrimaryKey="true"
            enableDeleteByExample="true"
        >
            <property name="immutable" value="false" />
            <property name="constructorBased" value="true" />
        </table>

    </context>
</generatorConfiguration>
```

Generated model files - Tbit88HpKey 

``` java
import java.io.Serializable;
public class Tbit88HpKey implements Serializable {
    private Long cSeg;
    private String cTypStr;
    private Integer nInt;
    private static final long serialVersionUID = 1L;
    public Tbit88HpKey(Long cSeg, String cTypStr, Integer nInt) {
        this.cSeg = cSeg;
        this.cTypStr = cTypStr;
        this.nInt = nInt;
    }
    // ...
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append(getClass().getSimpleName());
        sb.append(" [");
        sb.append("Hash = ").append(hashCode());
        sb.append(", cSeg=").append(cSeg);
        sb.append(", cTypStr=").append(cTypStr);
        sb.append(", nInt=").append(nInt);
        sb.append(", serialVersionUID=").append(serialVersionUID);
        sb.append("]");
        return sb.toString();
    }
}
```

Generated model files - Tbit88Hp

``` java
import java.io.Serializable;
import java.util.Date;
public class Tbit88Hp extends Tbit88HpKey implements Serializable {
    private Float qSpd;
    private Float qEstVar;
    private Float qRlb;
    private Date dLstUpd;
    private static final long serialVersionUID = 1L;
    public Tbit88Hp(Long cSeg, String cTypStr, Integer nInt, Float qSpd, Float qEstVar, Float qRlb, Date dLstUpd) {
        super(cSeg, cTypStr, nInt);
        this.qSpd = qSpd;
        this.qEstVar = qEstVar;
        this.qRlb = qRlb;
        this.dLstUpd = dLstUpd;
    }
    // ...
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append(getClass().getSimpleName());
        sb.append(" [");
        sb.append("Hash = ").append(hashCode());
        sb.append(", qSpd=").append(qSpd);
        sb.append(", qEstVar=").append(qEstVar);
        sb.append(", qRlb=").append(qRlb);
        sb.append(", dLstUpd=").append(dLstUpd);
        sb.append(", serialVersionUID=").append(serialVersionUID);
        sb.append("]");
        return sb.toString();
    }
}
```

Thanks
Luciano

**Comment by viruscamp** [comment]
<plugin type="org.mybatis.generator.plugins.ToStringPlugin">
    <property name="useToStringFromRoot" value="true" />
</plugin>

**Comment by TroyTae** [comment]
@caladyon 
Additional this bug fixed until 1.3.3 version.
How about upgrade your MBG version in the `pom.xml`.

**Comment by jeffgbutler** [comment]
Assuming this is resolved via the `useToStringFromRoot` setting.
